massa-network/src/node_worker.rs

```diff
@@ -179,17 +179,47 @@ impl NodeWorker {
         })?;
         let write_timeout = self.cfg.message_timeout;
         let node_id_copy = self.node_id;
+        let storage = self.storage.clone();
         let node_writer_handle = tokio::spawn(async move {
             loop {
                 match writer_command_rx.recv().await {
                     Some(to_send) => {
                         let bytes_vec: Vec<u8> = match to_send {
                             ToSend::Msg(msg) => msg.to_bytes_compact().unwrap(),
                             ToSend::Block(block_id) => {
-                                break;
+                                // TODO: precisely allocate?
```

Comment:
we know a block is at most max_block_size. And when all blocks will be full it will be exactly that.